File: api/src/main/java/me/kvdpxne/dtm/team/Team.java
package me.kvdpxne.dtm.team;

import java.util.UUID;
import me.kvdpxne.dtm.capabilities.Identifiable;
import me.kvdpxne.dtm.capabilities.Nameable;
import me.kvdpxne.dtm.team.color.TeamColor;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @since 0.1.0
 */
public interface Team
  extends
  Identifiable<UUID>,
  Nameable {

  /**
   * @since 0.1.0
   */
  @Override
  @NotNull
  UUID getIdentifier();

  /**
   * @since 0.1.0
   */
  @Override
  @NotNull
  String getName();

  /**
   * @since 0.1.0
   */
  @NotNull
  TeamColor getColorOfArmor();

  /**
   * @since 0.1.0
   */
  @NotNull
  TeamColor getColorOfProfession();

  /**
   * @since 0.1.0
   */
  @NotNull
  TeamColor getColorOnChat();

  /**
   * @since 0.1.0
   */
  @Nullable
  TeamColor getColorOnPlayerList();

  /**
   * @since 0.1.0
   */
  @NotNull
  LocalTeam toLocalTeam();
}

File: src/main/kotlin/me/kvdpxne/dtm/team/Team.kt
package me.kvdpxne.dtm.team

import me.kvdpxne.dtm.game.Game
import me.kvdpxne.dtm.shared.Identifiable
import me.kvdpxne.dtm.shared.Nameable
import me.kvdpxne.dtm.shared.TeamUuid
import org.bukkit.ChatColor
import org.bukkit.DyeColor

/**
 * @since 0.1.0
 */
interface Team : Identifiable<TeamUuid>, Nameable {

  /**
   * @since 0.1.0
   */
  val game: Game<out Team>

  /**
   * @since 0.1.0
   */
  val color: TeamColor

  /**
   * @since 0.1.0
   */
  val colorInChat: ChatColor
    get() = this.color.first

  /**
   * @since 0.1.0
   */
  val professionColor: ChatColor
    get() = this.color.second

  /**
   * @since 0.1.0
   */
  val dyeColor: DyeColor
    get() = this.color.third

  /**
   * @since 0.1.0
   */
  override val displayName: String
    get() = "${this.color.first}&l${this.name}"

  /**
   * @since 0.1.0
   */
  fun toLocalTeam(): LocalTeam
}